recipes/libhal/all/conanfile.py

```diff
@@ -36,14 +36,26 @@ def _compilers_minimum_version(self):
             "apple-clang": "14"
         }
 
+    @property
+    def _bare_metal(self):
+        return self.settings.os == "baremetal"
+
     def export_sources(self):
         export_conandata_patches(self)
 
     def layout(self):
         basic_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.83.0", transitive_headers=True)
+        # NOTE: from the author, kammce, to the conan team. boost-leaf is tied
+        # to the 2.0.1 and must be kept for it to work correctly.
+        version = Version(self.version)
+        if version == "2.0.1":
+            self.requires("boost-leaf/1.81.0",)
+        elif version == "2.0.2":
+            self.requires("boost/1.83.0",
+                          transitive_headers=True,
+                          options={ "header_only": True })
```

Comment:
The need of header_only, should be part of `validate()` too, because any user is able to override it by command line:

```
def validate(self):
    if not self.dependencies["boost"].options.header_only:
        raise ConanInvalidConfiguration(f"{self.ref} requires boost/*:header_only=True due boost::leaf.")
```

Reply:
Done